model/legacy-private/src/main/java/org/keycloak/migration/migrators/MigrateTo9_0_0.java

```diff
@@ -92,8 +97,13 @@ protected void addAccountConsoleClient(RealmModel realm) {
 
             client.setProtocol("openid-connect");
 
-            RoleModel role = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID).getRole(AccountRoles.MANAGE_ACCOUNT);
-            if (role != null) client.addScopeMapping(role);
+            ClientModel accountClient = realm.getClientByClientId(Constants.ACCOUNT_MANAGEMENT_CLIENT_ID);
+            if (accountClient != null) {
+                RoleModel role = accountClient.getRole(AccountRoles.MANAGE_ACCOUNT);
+                if (role != null) {
+                    client.addScopeMapping(role);
+                }
+            }
```

Comment:
This change may or may not be valid. File a bug report with justification before this PR.

Reply:
This really should be split into two issues/PRs

Reply:
Hi Hynek,

In fact, there is no bug reported for this change.

However, will occur a NPE if the keycloak administrator had removed the client 'account'. I can open a bug for this if you think better.

I believe, based on my observation, that the existence of this client should not interrupt the migration.

Reply:
> This really should be split into two issues/PRs

Ok

Reply:
Only for tracing.

#15020